Mlem/App/Views/Shared/ContentLoader.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  ContentLoader.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-13.
+//
+
+import Foundation
+import MlemMiddleware
+import SwiftUI
+
+struct ContentLoader<Content: View, Model: Upgradable>: View {
+    let model: any Upgradable
+    var content: (Model.MinimumRenderable) -> Content
+    
+    init(model: Model, content: @escaping (Model.MinimumRenderable) -> Content) {
+        self.model = model
+        self.content = content
+    }
+    
+    var body: some View {
+        if let modelValue = model.wrappedValue as? Model.MinimumRenderable {
+            content(modelValue)
+                .task {
+                    if !model.isUpgraded {
+                        await upgradeModel()
+                    }
+                }
```

Comment:
`.task` is executed whenever navigating back to the tab from another. Theoretically this could cause `upgradeModel` to be called more than once, if `.task` runs a second time whilst `upgradeModel` is in progress. A view-side boolean check would resolve this

Reply:
Nice catch! Fixed 👍